### PR TITLE
fix for issue 5013 : Ghatsad fixes and introduction of getVanaMidnight() function

### PR DESCRIPTION
**Pull request by waterlgndx**
https://github.com/DarkstarProject/darkstar/issues/5013

create a getVanaMidnight() function in common.lua, accurate within 3 earth seconds (each vana minute is 2.4 seconds)
replace item IDs with constant item names
rework logic for acquiring attachments
change so that players are required to wait until vana midnight instead of earth midnight.
apply style guide

### CODE REVIEW

**Comment by TeoTwawki** [comment]
We can't use the existing vanadiel time functions to check the day has changed?

File: scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua [hunk 1]
```diff
@@ -360,169 +226,169 @@ function onTrigger(player,npc)
...
-        player:startEventString(620, automatonName, automatonName, automatonName, automatonName, attachments, 0, 0, 0, 0, 2187, 1, unlockedAttachments);
-    elseif (NoStringsAttached == QUEST_COMPLETED and attachments == 3 and attachmentStatus == 0 and player:getMainJob() == dsp.job.PUP and player:getMainLvl() >= 40) then
-        player:startEventString(900, automatonName, automatonName, automatonName, automatonName, 0, 0, 0, 0, 0, 2186, 2);
-    elseif (NoStringsAttached == QUEST_COMPLETED and attachments == 4 and attachmentStatus == 0 and player:getMainJob() == dsp.job.PUP and player:getMainLvl() >= 50) then
-        if (unlockedAttachments == 30) then
-            player:startEventString(901, automatonName, automatonName, automatonName, automatonName, 1, 0, 0, 0, 0, 2186, 4);
-        elseif (unlockedAttachments == 46) then
-            player:startEventString(901, automatonName, automatonName, automatonName, automatonName, 0, 0, 0, 0, 0, 2186, 4);
+    if NoStringsAttached == QUEST_ACCEPTED and NoStringsAttachedProgress == 2 then
```
Comment (takhlaq) [inline scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua]: we dont need to keep checking if the quest is accepted, we can check it once, then check the progress inside the block that checks if its accepted

File: scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua [hunk 2]
```diff
@@ -360,169 +226,169 @@ function onTrigger(player,npc)
...
+        player:startEvent(262) -- he want you to go to Arrapago
+    elseif NoStringsAttached == QUEST_ACCEPTED and NoStringsAttachedProgress == 3 then
+        player:startEvent(263) -- reminder to go to Arrapago
+    elseif NoStringsAttached == QUEST_ACCEPTED and NoStringsAttachedProgress == 4 and Automaton == true then
+        player:startEvent(264) -- you give the antique automaton to him and need to wait a gameday
+    elseif NoStringsAttached == QUEST_ACCEPTED and NoStringsAttachedProgress == 5 and CreationReady == true then
+        player:startEvent(265) -- you go back for your automaton
+    elseif NoStringsAttached == QUEST_COMPLETED and attachments == 0 and attachmentStatus == 0 and player:getMainJob() == dsp.job.PUP and player:getMainLvl() >= 10 then
```
Comment (takhlaq) [inline scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua]: same here

File: scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua [hunk 3]
```diff
@@ -360,169 +226,169 @@ function onTrigger(player,npc)
...
+        player:startEventString(620, automatonName, automatonName, automatonName, automatonName, attachments, 0, 0, 0, 0, 2185, 3, unlockedAttachments)
+    elseif NoStringsAttached == QUEST_COMPLETED and attachments == 0 and attachmentStatus == 1 then
+        player:startEvent(621, 0, 0, 0, 0, 0, 2185, 3, unlockedAttachments)
+    elseif NoStringsAttached == QUEST_COMPLETED and attachments == 1 and attachmentStatus == 0 and player:getMainJob() == dsp.job.PUP and player:getMainLvl() >= 20 then
```
Comment (takhlaq) [inline scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua]: we can store the main job in a local variable here too

File: scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua [hunk 4]
```diff
@@ -360,169 +226,169 @@ function onTrigger(player,npc)
...
-    elseif (NoStringsAttached == QUEST_COMPLETED and attachmentStatus == 10 and attachmentReady == true) then
-        player:startEventString(627, automatonName, automatonName, automatonName, automatonName, 0, 3);
-    elseif (NoStringsAttached == QUEST_COMPLETED and attachmentStatus == 2) then
-        if (attachments == 0) then
-            player:startEvent(622, 0, 1, 0, 0, 0, 2185, 3);
-        elseif (attachments == 1) then
-            player:startEvent(622, 0, 1, 0, 0, 0, 2186, 3);
-        elseif (attachments == 2) then
-            player:startEvent(622, 0, 1, 0, 0, 0, 2187, 1);
+    elseif NoStringsAttached == QUEST_COMPLETED and attachments == 2 and attachmentStatus == 1 then
```
Comment (takhlaq) [inline scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua]: here too

File: scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua [hunk 5]
```diff
@@ -360,169 +226,169 @@ function onTrigger(player,npc)
...
-    elseif (NoStringsAttached == QUEST_COMPLETED and (attachmentStatus == 12 or attachmentStatus == 13) and attachmentReady == true and attachmentWait > 0) then
-        player:startEvent(903, attachmentWait, 0);
-    elseif (NoStringsAttached == QUEST_COMPLETED and (attachmentStatus == 12 or attachmentStatus == 13) and attachmentReady == true and attachmentWait == 0) then
-        player:startEvent(905, attachmentStatus-12);
-    elseif (NoStringsAttached == QUEST_COMPLETED and attachmentStatus == 14) then
-        if (unlockedAttachments == 30) then
-            player:startEventString(901, automatonName, automatonName, automatonName, automatonName, 1, 0, 1, 0, 0, 2186, 4);
-        elseif (unlockedAttachments == 46) then
-            player:startEventString(901, automatonName, automatonName, automatonName, automatonName, 0, 0, 1, 0, 0, 2186, 4);
+    elseif NoStringsAttached == QUEST_COMPLETED and attachmentStatus == 11 and attachments == 3 then
```
Comment (takhlaq) [inline scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua]: here too

File: scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua [hunk 6]
```diff
@@ -360,169 +226,169 @@ function onTrigger(player,npc)
...
+        player:startEventString(900, automatonName, automatonName, automatonName, automatonName, 0, 0, 1, 0, 0, 2186, 2)
+    elseif NoStringsAttached == QUEST_COMPLETED and (attachmentStatus == 12 or attachmentStatus == 13) and attachmentReady == false then
```
Comment (takhlaq) [inline scripts/zones/Aht_Urhgan_Whitegate/npcs/Ghatsad.lua]: can group stuff below this under a single check for quest completed and attachment status, then the rest can be checked

**Comment by waterlgndx** [comment]
@TeoTwawki yeah we could do that if you'd prefer that.  The only reason I did it this way was because that thought never occurred until after I made all the changes.  I think it takes two player variables to do it that way but I think I could use `CreationStarted_Day` and `CreationStarted_Year` to achieve that.  I wonder if it should be something like PUP_CreationStarted_Day or something like that.

**Comment by TeoTwawki** [comment]
Eh, I'll leave it to others to tell you if they want it changed instead of a new function. I just thought we had that functionality somehwere already in existing scripts (and couldn't look, at work)

**Comment by waterlgndx** [comment]
@takhlaq sorry I missed that last one.   (quest completed and attachment status 12 or 13)

**Pull request merged**